### ISSUE TITLE
Performance issues and slow responsiveness browser extension

### ISSUE THREAD

**Issue by lazylizardv2** [bitwarden/browser#2335]
### Steps To Reproduce

1. Go to Bitwarden Plugin Icon
2. Click on the Icon
3. And wait...........
4. It will eventually load but is extremely slow to load vault entries and autofill is just as slow

### Expected Result

Older version of either bitwarden or vivaldi this process would be more or less instant

### Actual Result

Opening vault, viewing items, copying autofill all extremely slow, almost like plugin has crashed

### Operating System

macOS

### Operating System Version

Big Sur 11.6 (20G165)

### Web Browser

Vivaldi

### Browser Version

5.1.2567.39

### Build Version

Version: 1.56.0

**Comment by lazylizardv2** [comment]
I just get the spinning dots of doom...
![Screen Shot 2022-02-11 at 1 23 17 pm](https://user-images.githubusercontent.com/99458334/153533077-18d8aab1-679c-47ab-9923-72262fe2b394.png)
.

**Comment by cscharf** [comment]
Verified impacts to both Chromium and Mozilla browsers.

**Comment by kspearrin** [comment]
We have identified the issue and a hotfix should be incoming soon.

**Comment by cscharf** [comment]
Released and submitted to Chrome and Opera extension stores: https://github.com/bitwarden/browser/releases/tag/v1.56.1

**Comment by cscharf** [comment]
After testing of the initial fix, while we still believe we've identified the root cause, the fix caused other problems that we need to revisit a better implementation pattern on. Reopening this issue and will address it soon.

**Comment by SchneiderSam** [comment]
Thanks for working on this. Can't use BITWARDEN. In Chrome is totally crashed...

**Comment by cscharf** [comment]
> Thanks for working on this. Can't use BITWARDEN. In Chrome is totally crashed...

Do you have crash logs or anything you can share from this? The performance issues we're seeing are very different from a crash as they're simply caused by an over-reliance on I/O bound operations for some heavy traffic'd data; but it shouldn't cause a crash (that may be something different).

**Comment by SchneiderSam** [comment]
> Released and submitted to Chrome and Opera extension stores: https://github.com/bitwarden/browser/releases/tag/v1.56.1

Still not available. Can i install the extension manually?

**Comment by cscharf** [comment]
I would wait for v1.56.2 (about to be cut).

**Comment by cscharf** [comment]
@SchneiderSam , looks like you're being impacted by:
* #1310
* #2334
* and this one

The upcoming build/release won't help you as those items are still in development as of this morning 🙁 .

**Comment by dv336699** [comment]
Load the 1.55.0 for now and you should be good

https://github.com/bitwarden/browser/releases/tag/v1.55.0

**Comment by idkw** [comment]
@dv336699 How do you install it ?

**Comment by SchneiderSam** [comment]
@dv336699 THANKS!

**Comment by djprmf** [comment]
I have the most current version - 1.56 - and with the recent updates the extension is still using a lot of CPU from time to time, along with slow loading of the contents everytime.

![tugatech-2022-02-11-aeda6edd-56bf-464c-929b-4c0bbe219757](https://user-images.githubusercontent.com/6960243/153576049-62112221-7507-4903-9ddf-2e70eefaa62a.jpg)

This started to happen in the latest update of the extension.

**Comment by cscharf** [comment]
The slowness is due to heavy disk I/O as we changed some underlying storage mechanisms to facilitate account switching. This had an unfortunate side-effect on many people's machines in how it impacted latent disk reads. We are working on a good in-memory caching scheme to buffer or cache against these reads to ensure much better performance. 1.56.0, 1.56.1, and 1.56.2 will have these same performance issues.

**Comment by cscharf** [comment]
Apparently this made a large difference in performance, but it caused some other issues, so we're leaning in that direction, https://github.com/bitwarden/jslib/pull/668

**Comment by robwent** [comment]
I'm also on 1.56 and it's causing the whole browser to lock up.
Windows 11

I also saw this last night https://github.com/bitwarden/browser/issues/2335#issuecomment-1035958454
I thought I'd given the wrong password, but looking at that comment, it may have happened after I logged in with a Yubi key and then been fine again when I logged out and back in without needing it.

**Comment by brimwats** [comment]
I am also having entire browser lockups for bitwarden on Vivaldi (based on Chrome)

```
Vivaldi | 5.1.2567.39 (Stable channel) (64-bit)
-- | --
Revision | 8963a81d8c1c8fcb85102a26391fa62c3887f8cd
OS | Windows 11 Version 21H2 (Build 22000.493)
JavaScript | V8 9.8.177.9
User Agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.88 Safari/537.36
```

**Comment by pgsill** [comment]
Biometric unlock (with Touch ID) no longer works for me. Chrome 98, macOS 11.6, Bitwarden 1.30.0.

**Comment by cscharf** [comment]
We have a fix ready, going through internal QA now and we're hoping to have a release cut shortly. Testing bits can be found once this build is complete if you want an early preview: https://github.com/bitwarden/browser/actions/runs/1829545464

**Comment by kriswilk** [comment]
I see the hotfixes are coming fast and furious. 1.56.1 did improve performance.

Glad you (Bitwarden) are working to fix this properly but one has to wonder why this wasn't caught before general release. Methinks even a handful of testers would have been enough to send up red flags, no?

**Comment by cscharf** [comment]
> I see the hotfixes are coming fast and furious. 1.56.1 did improve performance.
> 
> 
> 
> Glad you (Bitwarden) are working to fix this properly but one has to wonder why this wasn't caught before general release. Methinks even a handful of testers would have been enough to send up red flags, no?

Our QA team is constantly evolving and frequently adds to the thousands of regression tests we go through each release. Every release is a learning experience, especially in cases like this where we've had such a hard time reproducing issues. There plenty of opportunities to cover in our retrospective this go around. Thank you for your patience and support!

**Comment by robwent** [comment]
v1.56.1 is working well for me now.

**Comment by tatiee** [comment]
v1.56.1 working for me as well

**Comment by rymancl** [comment]
This seems to be an active issue, so logging this here. Let me know if I should redirect elsewhere.

-----

I'm still getting the `[error: cannot decrypt]` in v1.56.1. Browser is Chrome.

![image](https://user-images.githubusercontent.com/3211227/153654968-b59982b5-21f5-45d7-912c-c17eb2a08d51.png)

![image](https://user-images.githubusercontent.com/3211227/153654924-a82c7fb1-f43b-4dc4-a5e4-d45dd3325833.png)

-----

If I log out, I'm not able to log back in.

![image](https://user-images.githubusercontent.com/3211227/153655367-97d574b4-5bc3-4837-a588-5c8ce0dfe001.png)

**Comment by rando-calrissian** [comment]
> ![image](https://user-images.githubusercontent.com/3211227/153655367-97d574b4-5bc3-4837-a588-5c8ce0dfe001.png)

I have this same issue with 1.56.1.

**Comment by djsmith85** [comment]
Closing this, as the issues are resolved as of version `1.56.3`

**Comment by saimon-moore** [comment]
It's happening again....
